diff/walking/differ.go

```diff
@@ -153,34 +166,47 @@ func (s *walkingDiff) DiffMounts(ctx context.Context, lower, upper []mount.Mount
 	}
 	defer mount.Unmount(bDir, 0)
 
-	cw, err := s.store.Writer(ctx, ref, 0, "")
+	if config.Reference == "" {
+		config.Reference = uniqueRef()
+	}
+
+	cw, err := s.store.Writer(ctx, config.Reference, 0, "")
 	if err != nil {
 		return emptyDesc, errors.Wrap(err, "failed to open writer")
 	}
 
-	var opts []content.Opt
 	if isCompressed {
 		dgstr := digest.SHA256.Digester()
 		compressed, err := compression.CompressStream(cw, compression.Gzip)
 		if err != nil {
+			cw.Close()
```

Comment:
any reason this can't be a defer?

Reply:
This is acting more like a cancel with the `Commit` also finalizes. I am trying to avoid calling `Close` after `Commit` because I think it is sloppy. It is more like `Rollback` in this case.

Reply:
I see, thanks for explaining. Is this documented somewhere?

Reply:
> Is this documented somewhere?

Duh it's documented, just those documents say something completely different than what I claimed. I am updating this to defer and call `Abort` when the reference is generated.